### PR TITLE
Users can now specify a username before connecting

### PR DESCRIPTION
**Pull request by stevenhillcox**

### CODE REVIEW

File: src/client/app/home/home.component.html [hunk 1]
```diff
@@ -1,7 +1,11 @@
 <div class="login-box">
-    <button (click)="guestSignIn()">
-        Join as guest
-    </button>
+    <form (ngSubmit)="guestSignIn()">
+        <input [(ngModel)]="userName" type="text" placeholder="Name (Optional)"/>
```
Comment (kenneth-gray) [inline src/client/app/home/home.component.html]: Think we should just one way bind this as we won't be setting it in the home component

File: src/client/app/home/home.component.spec.ts [hunk 2]
```diff
@@ -125,12 +125,18 @@ describe('HomeComponent', () => {
     });
 
     describe('guestSignIn', () => {
-        beforeEach(() => {
-            homeComponent.guestSignIn();
+
+        describe('It stores the userName', () => {
```
Comment (kenneth-gray) [inline src/client/app/home/home.component.spec.ts]: Is this actually running?

File: src/server/client/client-service.js [hunk 3]
```diff
@@ -19,10 +19,10 @@ module.exports = function(namingService, googleService) {
 
     return _clientService;
 
-    function createClient(token, websocket) {
+    function createClient(token, websocket, userName) {
         var deferred = Q.defer();
 
-        var client = new Client(websocket, _namingService, _clientService);
+        var client = new Client(websocket, _namingService, _clientService, userName);
```
Comment (kenneth-gray) [inline src/server/client/client-service.js]: We should probably make user name the first parameter

Comment (stevenhillcox) [inline src/server/client/client-service.js]: I disagree, userName here is the only optional parameter so I believe it should go at the end.

Comment (kenneth-gray) [inline src/server/client/client-service.js]: We do always pass it in though? Might tidy up in separate PR

**Pull request closed without merging**